Microsoft.NET.Build.Containers/ImageConfig.cs

```diff
@@ -29,53 +37,108 @@ internal ImageConfig(JsonNode config)
         _labels = GetLabels();
         _exposedPorts = GetExposedPorts();
         _environmentVariables = GetEnvironmentVariables();
+        _rootFsLayers = GetRootFileSystemLayers();
+        _architecture = GetArchitecture();
+        _os = GetOs();
+        _history = GetHistory();
     }
 
+    private List<HistoryEntry> GetHistory() => _config["history"]?.AsArray().Select(node => node.Deserialize<HistoryEntry>()!).ToList() ?? new List<HistoryEntry>();
+    private string GetOs() => _config["os"]?.ToString() ?? throw new ArgumentException("Base image configuration should contain an 'os' property.");
+    private string GetArchitecture() => _config["architecture"]?.ToString() ?? throw new ArgumentException("Base image configuration should contain an 'architecture' property.");
+
     /// <summary>
     /// Builds in additional configuration and returns updated image configuration in JSON format as string.
     /// </summary>
     internal string BuildConfig()
     {
-        if (_config["config"] is not JsonObject config)
+        var newConfig = new JsonObject();
```

Comment:
mutating the underlying config was making configs that were noisier than expected, and in a `scratch` container there _is_ no underlying config. It's more easy to think about creating an entirely new config object from first principles.

Reply:
Though this does require us to be explicit about pull through fields that we don't support like users, volumes, and stop signals